### PR TITLE
delete extra drpdown & alter drpdown js

### PR DESCRIPTION
**Pull request by bethanyhaubert**

### CODE REVIEW

**Comment by bethanyhaubert** [comment]
tyring to get onchange to work. looked at a few help pages. http://help.dottoro.com/ljoiurdq.php

**Comment by bethanyhaubert** [comment]
tried both of these:
https://developer.mozilla.org/en-US/docs/Web/Events/change
http://help.dottoro.com/ljoiurdq.php

File: app/models/client.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@ class Client < ActiveRecord::Base
    def recent_touches
     self.touches # TODO Make this only return one touch for the client.
    end
+
```
Comment (sumeetjain) [inline app/models/client.rb]: Remove trailing whitespace

File: app/controllers/pages_controller.rb [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+      @business.clients.each do |client|
+        if Touch.where({"client_id" => @current_client}) != nil
+          @clients << client
+        end
       end
+      return @clients
     end
   end
-
+
```
Comment (sumeetjain) [inline app/controllers/pages_controller.rb]: Remove trailing whitespace

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -1,18 +1,19 @@
 class PagesController < ApplicationController
+
```
Comment (sumeetjain) [inline app/controllers/pages_controller.rb]: Extra empty line detected at class body beginning.

File: app/controllers/pages_controller.rb [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
     else
-      @clients = @business.clients
-
-      if Touch.where({"client_id" => @current_client}) != nil
-        @touches = Touch.where({"client_id" => @current_client})
-      else
-        @touches = nil
+      @clients = []
+      @business.clients.each do |client|
+        if Touch.where({"client_id" => @current_client}) != nil
```
Comment (sumeetjain) [inline app/controllers/pages_controller.rb]: Prefer `!expression.nil?` over `expression != nil`.

File: app/controllers/pages_controller.rb (hunk 2 above)
Comment (sumeetjain) [inline app/controllers/pages_controller.rb]: Trailing whitespace detected.

File: app/models/client.rb (hunk 1 above)
Comment (sumeetjain) [inline app/models/client.rb]: Trailing whitespace detected.

File: app/assets/javascripts/display_selected_client_from_dropdown.js [hunk 5]
```diff
@@ -3,24 +3,30 @@ jQuery(document).ready(function(){
 
   parent.addEventListener("click", function(event){
 
-    if (event.target.matches(".dropdown_submit")){
-      var selectedId = document.getElementById("dropdown_select");
-      var presentation_area = document.getElementById("current_client");
+    // if (event.target.matches(".dropdown_submit")){
+    if (event.target.matches("dropdown_select")){
```
Comment (sumeetjain) [inline app/assets/javascripts/display_selected_client_from_dropdown.js]: This needs to be `event.target.matches("#dropdown_select")` if it's finding something with an ID of `dropdown_select`. Doing `matches("dropdown_select")` means you're looking for an element with a **tag** called `<dropdown_select>`.

**Comment by bethanyhaubert** [comment]
ok, added the '#'.  still getting error. https://www.dropbox.com/s/olrt12fzs57i1wo/Screenshot%202016-03-31%2014.04.47.png?dl=0

**Comment by bethanyhaubert** [comment]
i don't get how
 ```
function wordsWordsWords (more_words) {
}
```
should be used

**Comment by sumeetjain** [comment]
Is some of this copied from a tutorial or something? I think it might not be 100% compatible with your goal. Can you try removing the `function OnSelectionChange (select)` scope entirely?

And then I think `var selectedOption = select.options[select.selectedIndex];` would be `var selectedOption = event.target.options[select.selectedIndex];`

**Comment by bethanyhaubert** [comment]
still not working

**Comment by sumeetjain** [comment]
`if (event.target.matches(".dropdown"))` is looking for elements that have a **class** of "dropdown". Is that your intention? I thought you were trying to check for clicks occurring on `<select id="dropdown_select"`...>`. That's an **ID**, so you'd need `if (event.target.matches("#dropdown"))`.

Secondly, delete ` onchange="OnSelectionChange (this)"` from the `<select>` tag. It's an outdated technique that's been replaced by `addEventListener`.

**Comment by bethanyhaubert** [comment]
can you just tell me what's wrong? i've changed something each time you've given feedback and i've changed it so much there's no way it's anywhere near what it was originally and i'm over it.

**Pull request merged**